crates/rolldown_common/src/module/mod.rs

```diff
@@ -1 +1,94 @@
 pub mod external_module;
+
+use crate::{EcmaModule, ExternalModule, ModuleIdx};
+
+#[derive(Debug)]
+pub enum Module {
```

Comment:
I didn't use trait here, because we don't have the needs to support custom modules from outside.

Another reason it that exhaustive matching force us to think the relation ship between each module type.